proxy/hdrs/MIME.cc

```diff
@@ -2561,8 +2561,8 @@ mime_parser_parse(MIMEParser *parser, HdrHeap *heap, MIMEHdrImpl *mh, const char
     field_value.ltrim_if(&ParseRules::is_ws);
     field_value.rtrim_if(&ParseRules::is_wslfcr);
 
-    // Make sure the name or value is not longer than 64K
-    if (field_name.size() >= UINT16_MAX || field_value.size() >= UINT16_MAX) {
+    // Make sure the name or value is not longer than configured max_hdr_field_size
+    if (field_name.size() >= max_hdr_field_size || field_value.size() >= max_hdr_field_size) {
```

Comment:
I understand this check had to keep the compatibility, but "header field" is defined on rfc7230 as:
```
 header-field   = field-name ":" OWS field-value OWS
```

So technically this check is incorrect because the setting name is max *header field* size. We may want to change the name or the check because it can cause confusion.

Reply:
Do we want to check header name and value separately? If we are going to keep it, I have no idea for the setting name.

Maybe we can loosen the limitation so the check matches the setting name. How about having 128k as the default value of `max_hdr_field_size ` and checking the size by `name.size() + value.size() >= max_hdr_field_size`?

I know 128k sounds too large, but all headers we have been accepted should be covered with it. Or we can tighten it on 9.0.

Reply:
Refactored to use 128K default. Does this look better?

Reply:
Discussed with Masakaju and updated the patch accordingly.